SimPEG/base/pde_simulation.py

```diff
@@ -232,10 +263,21 @@ def MfDeriv_prop(self, u, v=None, adjoint=False):
                 return Zero()
             stash_name = f"_Mf_{arg}_deriv"
             if getattr(self, stash_name, None) is None:
-                M_prop_deriv = self.mesh.get_face_inner_product_deriv(
-                    np.ones(self.mesh.n_cells)
-                )(np.ones(self.mesh.n_faces)) * getattr(self, f"{arg.lower()}Deriv")
-                setattr(self, stash_name, M_prop_deriv)
+                prop = getattr(self, arg.lower())
+                t_type = TensorType(self.mesh, prop)
+
+                M_func = self.mesh.get_face_inner_product_deriv(model=prop)
```

Comment:
a question / thought on naming... `M_func` doesn't prompt me to think about derivatives. Would `M_deriv_func` work? It is the function of getting the derivative of the face inner product matrix ?